pkg/reference/ref.go

```diff
@@ -0,0 +1,176 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reference
```

Comment:
Nit: Maybe this should be nested one level deeper since I think we plan on adding additional packages. Something like `pkg/utils/reference` or `pkg/machinery/reference`?

Reply:
👍 I'm good with machinery.